src/swarm-tui/Swarm/TUI/Model/Repl.hs

```diff
@@ -275,6 +278,7 @@ data REPLState = REPLState
   { _replPromptType :: REPLPrompt
   , _replPromptEditor :: Editor Text Name
   , _replValid :: Bool
+  , _replErrorSrcLoc :: SrcLoc
```

Comment:
Right now we have `replValid :: Bool` and `replErrorSrcLoc :: SrcLoc`; but as far as I understand there can only be a `SrcLoc` other than `NoLoc` when `replValid` is `False`.  So why not combine them into a single field like `replValid :: Either SrcLoc ()` ?

Reply:
Sure, it's almost isomorphic and should be easy to refactor. 👍